src/tokenized-buffer.coffee

```diff
@@ -126,9 +125,17 @@ class TokenizedBuffer extends Model
     @disposables.add(@configSubscriptions)
 
     @retokenizeLines()
-    @packageManager.triggerActivationHook("#{grammar.packageName}:grammar-used")
+
     @emitter.emit 'did-change-grammar', grammar
 
+    # Delay this to the next tick to ensure whoever created the buffer has the
+    # chance to listen for this event before we send it.
+    process.nextTick =>
```

Comment:
Or maybe `TokenizedBuffer` could emit the current grammar when someone subscribes?